apps/server/src/controllers/userControllers/wallets/walletBalances.ts

```diff
@@ -22,49 +22,35 @@ export const getUserWalletBalances = async (req: RequestExt, res: Response) => {
       include: [
         {
           model: Groups,
-          where: { ownerId: userId },
-          as: 'group',
           include: [
             {
               model: GroupWallet,
-              as: 'groupWallet',
               attributes: ['balance'],
             },
           ],
         },
       ],
     })
 
-    if (personalWallets && groupBalances) {
-      const personal = personalWallets.reduce(
-        (acc, wallet) => {
-          acc[wallet.type] = {
-            id: wallet.id,
-            balance: wallet.balance,
-          }
-          return acc
-        },
-        {} as Record<string, unknown>
-      )
-
+    if (personalWallet && groupBalances) {
       const groups = (
         groupBalances as unknown as Record<string, unknown>[]
       ).reduce(
         (acc, cur) => {
-          const group: any = cur.group
-          acc.balance += group.groupWallet.balance
+          const group: any = cur.Group
```

Comment:
here you are dealing with data that is eagerLoaded from the Db which is only possible  because of the association specified between the models. so you can't explicitly specify a type for it.

Reply:
The compiler effectively treats **any** as “please turn off type checking for this thing”. It is similar to putting an @ts-ignore comment around every usage of the variable. This can be very helpful when you are first migrating a JavaScript project to TypeScript as you can set the type for stuff you haven’t migrated yet as any.

In cases where you don’t know what type you want to accept, or when you want to accept anything because you will be blindly passing it through without interacting with it, you can use [unknown](https://www.typescriptlang.org/play/#example/unknown-and-never).

Therefore in this case, since you know what you are expecting you can create an expected type then replace:

```js
groupBalances as unknown as YourExpectedType[]
```